jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -60,21 +60,28 @@ static ecma_value_t
 ecma_builtin_regexp_prototype_flags_helper (ecma_value_t this, /**< this value */
                                             uint16_t *flags_p) /**< [out] flags */
 {
-  if (!ecma_object_is_regexp_object (this))
+  if (ecma_is_value_object (this))
   {
-    return ecma_raise_type_error (ECMA_ERR_MSG ("Incompatible type"));
-  }
-
-  ecma_extended_object_t *re_obj_p = (ecma_extended_object_t *) ecma_get_object_from_value (this);
-  re_compiled_code_t *bc_p = ECMA_GET_INTERNAL_VALUE_ANY_POINTER (re_compiled_code_t,
-                                                                  re_obj_p->u.class_prop.u.value);
+    ecma_object_t *obj_p = ecma_get_object_from_value (this);
 
-  if (bc_p != NULL)
-  {
-    *flags_p = bc_p->header.status_flags;
+    if (ecma_get_object_type (obj_p) == ECMA_OBJECT_TYPE_CLASS)
+    {
+      ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) obj_p;
+      if (ext_obj_p->u.class_prop.class_id == LIT_MAGIC_STRING_REGEXP_UL
+          || ext_obj_p->u.class_prop.class_id == LIT_MAGIC_STRING_REGEXP_MATCHER)
+      {
```

Comment:
Would it be possible to have a function which does this check? ( And optionally returns the extended object?)

Reply:
Do you mean like this?
```.js
static bool
ecma_op_check_object_type_is_class (ecma_value_t this, /**< this value */
                                    ecma_extended_object_t **ext_obj_p) /**< [out] pointer */
{
  if (ecma_is_value_object (this))
  {
    ecma_object_t *obj_p = ecma_get_object_from_value (this);

    if (ecma_get_object_type (obj_p) == ECMA_OBJECT_TYPE_CLASS)
    {
      *ext_obj_p = (ecma_extended_object_t *) obj_p;
      return true;
    }
  }

  return false;
}
```